src/main/java/com/statsig/androidsdk/Store.kt

```diff
@@ -44,8 +44,8 @@ internal class Store (private var userID: String?, private var customIDs: Map<St
         cacheById = mutableMapOf()
 
         if (cachedResponse != null) {
-            val type = object : TypeToken<MutableMap<String, Cache>>() {}.type
             try {
+                val type = object : TypeToken<MutableMap<String, Cache>>() {}.type
```

Comment:
a little concerned about this; is there a way we can also alert the developer that the latest values won't be reflected in this case?

Reply:
to be a little more clear: one thing I think is really helpful is to communicate in cases when the library's behavior is going to be different from what the client might expect. in this case, we'd expect the most recent value to be updated if the app doesn't crash or raise an exception. but here, the error is swallowed and the cache is simply not updated, without letting the user know their expectation of the library's behavior was incorrect. most users won't understand why their cache hasn't been updated until they look a few stackframes down and see this error being swallowed.

Reply:
In this case, its not the latest values that would be lost if this threw an exception - its the cached values.  Its a good point though that issues here will silently fail, yet cached values will never be available.

I think I'll add an error boundary here - we log exceptions that happen to a separate endpoint so we can debug sdk issues like this

Reply:
nice! that sounds great. in _that_ case... would you ever consider exposing callbacks for developers who want to see those errors surfaced in dev?

like, I'd love to say something like:

```
Statsig.onError { throwable -> if (BuildConfig.isDebug) throw throwable }
```

Reply:
yes, definitely.  We'll talk through how to expose that